### PR TITLE
add publish workflow

### PR DESCRIPTION
**Pull request by CaselIT**
Create a tag and it gets pushed.

See https://test.pypi.org/project/zzzeeksphinx/

### CODE REVIEW

**Comment by CaselIT** [comment]
To Do before this works. Update the secret `pypi_token` with one that can push to `zzzeeksphinx`

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: Publish tag
+
+on:
+  # run when a pushing a tag
+  push:
+    tags:
+      - '*'
```
Comment (zzzeek) [inline .github/workflows/publish.yaml:7]: can this be like "rel_?_?" so it's a little more specific ?

Comment (CaselIT) [reply .github/workflows/publish.yaml:7]: This is the tag name. usually are they not in the form `v?.?.?` ? like `v1.2.3`?

**Comment by CaselIT** [comment]
> 
> 
> To Do before this works. Update the secret `pypi_token` with one that can push to `zzzeeksphinx`

also remember to update the secret in the github setting. Currently there is one for the test pypi (I think)

File: .github/workflows/publish.yaml (hunk 1 above)
Comment (zzzeek) [reply .github/workflows/publish.yaml:7]: they are!   but for 15 years I've been using rel_X_Y and im old and tired

Comment (CaselIT) [reply .github/workflows/publish.yaml:7]: ok, so I'll change the tag here to `v*` (`v?.?.?` does not work for >10 numbers like `v1.13.1`)

Comment (zzzeek) [reply .github/workflows/publish.yaml]: can we add a second regexp for rel_ beacuse I am not going to be able to keep track of two different tagging formats in my projects:

```
tags: 
   - 'rel_*'
```

Comment (CaselIT) [reply .github/workflows/publish.yaml]: my bad. I'm looking at the other project and they are all `rel_x_y_z`. Will update

**Comment by CaselIT** [comment]
@zzzeek ping :)

**Comment by zzzeek** [comment]
ok here's my thing with this.

The way it is now, i want a new release, I do:

    python setup.py sdist
    twine upload dist/<file>

the worflow here suggests it is instead:

   git tag rel_x_y  # I have to know what x_y is
   git push github --tags

it doesnt seem any easier to me :)

**Comment by CaselIT** [comment]
Since currently there aren't tags in this project, it may be unnecessary.

Close this if you prefer.

I may tweak this a bit and add it to alembic, since there would be useful

**Pull request merged**